src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -316,6 +316,51 @@ static RefKind getRefKind(ParamInfo<TypeSymbol> param, ImmutableArray<CustomModi
             }
         }
 
+        internal void AddNullableTransforms(ArrayBuilder<byte> transforms)
+        {
+            ReturnTypeWithAnnotations.AddNullableTransforms(transforms);
+            foreach (var param in Parameters)
+            {
+                param.TypeWithAnnotations.AddNullableTransforms(transforms);
+            }
+        }
+
+        internal FunctionPointerMethodSymbol ApplyNullableTransforms(byte defaultTransformFlag, ImmutableArray<byte> transforms, ref int position)
+        {
+            bool madeChanges = ReturnTypeWithAnnotations.ApplyNullableTransforms(defaultTransformFlag, transforms, ref position, out var newReturnType);
+            var newParamTypes = ImmutableArray<TypeWithAnnotations>.Empty;
+            if (!Parameters.IsEmpty)
+            {
+                var paramTypesBuilder = ArrayBuilder<TypeWithAnnotations>.GetInstance(Parameters.Length);
+                bool madeParamChanges = false;
+                foreach (var param in Parameters)
+                {
+                    madeParamChanges |= param.TypeWithAnnotations.ApplyNullableTransforms(defaultTransformFlag, transforms, ref position, out var newParamType);
+                    paramTypesBuilder.Add(newParamType);
+                }
+
+                if (madeParamChanges)
+                {
+                    newParamTypes = paramTypesBuilder.ToImmutableAndFree();
+                    madeChanges = true;
+                }
+                else
+                {
+                    paramTypesBuilder.Free();
+                    newParamTypes = ParameterTypesWithAnnotations;
```

Comment:
>ParameterTypesWithAnnotations [](start = 36, length = 29)

Would it make sense to cache this in a local and iterate over this array in the foreach above?

Reply:
I didn't want to do this because it's possible that `ParameterTypesWithAnnotations` has not been realized yet, so I only wanted to access it if required.

---
In reply to: [420404877](https://github.com/dotnet/roslyn/pull/43961#discussion_r420404877) [](ancestors = 420404877)

Reply:
I actually considered going a step further and just immediately returning `this` if no changes to the return type was made here, but I felt like that made the code too clever.

---
In reply to: [420447100](https://github.com/dotnet/roslyn/pull/43961#discussion_r420447100) [](ancestors = 420447100,420404877)